integration_tests/src/main/python/conftest.py

```diff
@@ -62,6 +62,29 @@ def is_databricks_runtime():
 def is_emr_runtime():
     return runtime_env() == "emr"
 
+_is_nightly_run = False
+_is_precommit_run = False
+
+def is_nightly_run():
+    return _is_nightly_run
+
+def is_at_least_precommit_run():
+    return _is_ngihtly_run or _is_precommit_run
+
+def require_nightly_run_tests(description):
```

Comment:
Nit: IMO this function name doesn't convey that it will skip the test.  Maybe because I'm thinking it acts like `require` from Scala.  Maybe `skip_unless_nightly_tests` would be more explicit?